app/src/main/java/com/example/poste/models/Post.java

```diff
@@ -67,6 +76,24 @@ public String getUrl() {
      */
     public String getId() { return id; }
 
+    /**
+     * Deletes this post from the data model and backend
+     */
+    public void deletePost(Folder parentFolder) {
+        Post delPost = this;
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.deletePost(new DeletePost(delPost.getId()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                parentFolder.getPosts().remove(delPost);
```

Comment:
Remember that a response can indicate failure (such as 404, 500, etc) -- would be useful to handle these error responses gracefully

Reply:
Yes. In due time. From what we discussed though I'm focusing on the happy day cases first. for the sake of moving toward MVP.

Reply:
Roger. I would ask that you at least introduce a simple `if (response.isSuccessful()) { ... } else { ... }` block. While I understand building towards 'happy day' cases, we want to ensure we aren't needlessly introducing hard-to-troubleshoot bugs (where it appears things were successful, with no logging or display otherwise, even if they weren't)

Reply:
Added base "if (responce.isSuccessful()) {} else {} " branches and Toast messages to provide basic error handling,